src/histories/Branches.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Branches History
+ * Stores transaction history as a tree.
+ */
+
+import Tree from '../Tree'
+
+let Branches = function (options={}) {
+  this.base = {}
+  this.tree = new Tree()
+  this.maxHistory = options.maxHistory || -Infinity
```

Comment:
By default, I never want to keep any history. This feels more permanent than `0`.

Also I never miss a chance to use Infinity :)